client/src/test/java/io/oxia/client/session/SessionTest.java

```diff
@@ -125,6 +125,22 @@ void sessionId() {
         assertThat(session.getSessionId()).isEqualTo(sessionId);
     }
 
+    @Test
+    public void nonCallbackListener() {
+        final OxiaStubProvider mockProvider = mock(OxiaStubProvider.class);
+        final SessionNotificationListener listener = spy(SessionNotificationListener.class);
+        when(mockProvider.getStubForShard(anyLong()))
+                .thenThrow(new IllegalStateException("wrong states"));
+        var session =
+                new Session(
+                        executor, mockProvider, config, shardId, sessionId, InstrumentProvider.NOOP, listener);
+        try {
+            session.close();
```

Comment:
It's better to assert that the returend future should be failed with the expected exception.